tests/models_tests/models_test.go

```diff
@@ -790,6 +935,70 @@ func TestRetrieveHistogramDataBySourceIdAndConceptId(t *testing.T) {
 	}
 }
 
+func TestTransformDataIntoTempTable(t *testing.T) {
+	// This test checks whether the TransformDataIntoTempTable successfully
+	// caches temp tables based on query definitions
+
+	filterConceptDef :=
+		utils.CustomConceptVariableDef{
+			ConceptId: 2000006885,
+			Filters: []utils.Filter{
+				{
+					Type:  ">=",
+					Value: utils.Float64Ptr(1.0),
+				},
+			},
+			Transformation: "z_score",
+		}
+	omopDataSource := tests.GetOmopDataSource()
+	resultsDataSource := tests.GetResultsDataSource()
+
+	// run a simple query:
+	querySQL := "(SELECT person_id, observation_concept_id, value_as_number FROM " + omopDataSource.Schema + ".observation_continuous) as tmpTest "
+	query := resultsDataSource.Db.Table(querySQL)
+
+	tmpTableName1, _ := models.TransformDataIntoTempTable(omopDataSource, query, filterConceptDef)
+	// repeat the exact same query...it should return the same temp table:
+	tmpTableName2, _ := models.TransformDataIntoTempTable(omopDataSource, query, filterConceptDef)
+	if tmpTableName1 != tmpTableName2 {
+		t.Errorf("tmp table should have been reused")
+	}
+	// do a slightly different query...and the temp table should be a different one:
+	querySQL = "(SELECT person_id, observation_concept_id, value_as_number FROM " + omopDataSource.Schema + ".observation_continuous) as tmpTest2 "
+	query = resultsDataSource.Db.Table(querySQL)
+	tmpTableName3, _ := models.TransformDataIntoTempTable(omopDataSource, query, filterConceptDef)
+	if tmpTableName1 == tmpTableName3 {
+		t.Errorf("tmp table should have a new one")
+	}
+
+	// test log transformation
+	filterConceptDef =
+		utils.CustomConceptVariableDef{
+			ConceptId: 2000006885,
+			Filters: []utils.Filter{
+				{
+					Type:  ">=",
+					Value: utils.Float64Ptr(1.0),
+				},
+			},
+			Transformation: "log",
+		}
+	querySQL = "(SELECT person_id, observation_concept_id, value_as_number FROM " + omopDataSource.Schema + ".observation_continuous) as tmpTest3 "
+	query = resultsDataSource.Db.Table(querySQL)
+	tmpTableName4, _ := models.TransformDataIntoTempTable(omopDataSource, query, filterConceptDef)
+	if tmpTableName4 == "" {
```

Comment:
one suggestion is to add check the transformed table has correct number of rows and data value checks for particular row(s).

Reply:
this is indirectly already done in `TestRetrieveHistogramDataBySourceIdAndCohortIdAndConceptDefsPlusCohortPairs` 

But I agree some more specific tests can be added here. Especially to assert that the `TransformDataIntoTempTable` only applies transformations and not the filters themselves.

Reply:
done